### PR TITLE
Yaml templates sharing container image

### PR DESCRIPTION
**Pull request by Abacn**
Part of #2279

### CODE REVIEW

File: plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java [hunk 1]
```diff
@@ -649,37 +670,35 @@ protected String stageFlexTemplate(
...
+      File indexFile =
+          new File(
+              outputClassesDirectory.getPath() + "/" + containerName, PYTHON_LAUNCHER_YAML_INDEX);
+      try (FileWriter writer = new FileWriter(indexFile)) {
+        writer.write(containerStageTracker.getMappingJson(containerName));
       }
-      Path destination = Paths.get(outputClassesDirectory.getAbsolutePath(), "template.yaml");
-
-      LOG.info("Copying " + source + " to " + destination);
-      Files.copy(source, destination, StandardCopyOption.REPLACE_EXISTING);
```
Comment (Abacn) [inline plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java:680]: we do not to copy if not renaming as it already appears in the `target/classes` as java resource files

**Comment by Abacn** [comment]
R: @derrickaw

**Review by derrickaw — DISMISSED** [review]

**Comment by Abacn** [comment]
The latest commit fixed an edge case preventing container compatibility check working when a compatible template came first then an incompatible one reuse container (not happening in current Templates).

KafkaToIceberg Yaml test failed with managed transform launcher fail, a combination of b/479594634 (serverless launcher failing) and stock out of the region (GCE launcher cannot be provisioned) caused the issue, not related to this change

```
Dataflow is unable to process the managed transform(s) ref_AppliedPTransform_WriteToIceberg-SchemaAwareExternalTransform-beam-transform-managed-v1_6(beam:schematransform:org.apache.beam:iceberg_write:v1),ref_AppliedPTransform_ReadFromKafka-beam-schematransform-org-apache-beam-kafka_read-v1_3(beam:schematransform:org.apache.beam:kafka_read:v1) due to an internal error. Because the upgrade failed, the new job or the update request has been rejected. To troubleshoot, see https://cloud.google.com/dataflow/docs/guides/common-errors#managed-transforms-upgrade.
2026-01-29 20:47:08.047 EST
Unknown error in operation 'operation-1769737619256-6499126243540-62553c5b-cc61bf89': [ZONE_RESOURCE_POOL_EXHAUSTED_WITH_DETAILS] 'The zone 'projects/cloud-teleport-testing/zones/us-west2-b' does not have enough resources available to fulfill the request.  '(resource type:compute)'.'
```

update: second attempt failed one test, same reason:

```
Unknown error in operation 'operation-1769804533087-649a0ba843b9c-cc99d86b-07e149cc': [ZONE_RESOURCE_POOL_EXHAUSTED_WITH_DETAILS] 'The zone 'projects/cloud-teleport-testing/zones/us-west2-b' does not have enough resources available to fulfill the request. 
```

**Comment by Abacn** [comment]
It appears this PR triggered a weird bug of template parameter corruption: https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/21535675203

```
"details": [
    {
      "@type": "type.googleapis.com/google.dataflow.v1beta3.InvalidTemplateParameters",
      "parameterViolations": [
        {
          "description": "Conditional parameter had an invalid trigger value: 'AVRO_BINARY_ENCODING'",
          "parameter": "schemaFormat"
        }
      ]
    }
  ],
```

checking the template json, this PR caused json fields ordered alphabetically. 

Previously

```
{
        "name": "readBootstrapServerAndTopic",
        "groupName": "Source",
        "label": "Source Kafka Topic",
        "helpText": "Kafka Topic to read the input from.",
        "parentTriggerValues": [
          ""
        ],
        "paramType": "KAFKA_READ_TOPIC"
      },
```

now

```
{
                "groupName": "Source",
                "helpText": "Kafka Topic to read the input from.",
                "label": "Source Kafka Topic",
                "name": "readBootstrapServerAndTopic",
                "paramType": "KAFKA_READ_TOPIC",
                "parentTriggerValues": [
                    ""
                ]
            },
```

This is due to a line added during debugging forgot to remove: https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/3268/changes#diff-e974a7cb17b5eaece9f4bf8e4c0952d47a5de1313b51a8ac51fe0e872c3240f9R598

causing template spec file not overriden (and latter steps, e.g. SBOM/scanning skipped. Though there were not enabled in test run either)

This should still work as templates is built via `gcloud dataflow flex-template build` command as we are not actively changing template request proto recently. However it generates a invalid json seen by router. We should revisit it afterwards.

EDIT: reverting this PR the test failure persists: go/templates-pr/3286
